data-processing/entities/definitions/commands/detect_definitions.py

```diff
@@ -107,28 +134,263 @@ def get_term_definition_pairs(
-
-        term_start = min([r.start for r in term_range_list])
-        term_end = max([r.end for r in term_range_list]) + 1
-        definition_start = min([r.start for r in definition_range_list])
-        definition_end = max([r.end for r in definition_range_list]) + 1
+        term_a_rangelist = terms[pair_index]
+        definition_a_rangelist = definitions[pair_index]
+        term_confidence = term_confidences[pair_index]
+        definition_confidence = definition_confidences[pair_index]
+
+        term_start = min([r.start for r in term_a_rangelist])
+        term_end = max([r.end for r in term_a_rangelist]) + 1
+        definition_start = min([r.start for r in definition_a_rangelist])
+        definition_end = max([r.end for r in definition_a_rangelist]) + 1
+
+        # Decide types of terms and definitions.
+        #  - Term types: [symbol, term, abbreviation, entity, acrnonym]
+        #  - definition types: [nickname, definition] for symbol, [definition] for protologism, [expansion] for protologism
+        # Currently, our model doesn't predict [acronym] as a term and its [expansion] as a definition. In case we detect an abbreviation in the detected term, we split it into [acroynm] and [expansion], and make a new pair of term and definintion.
+        term_type = "symbol" if "SYMBOL" in text[term_start:term_end] else "term"
+        definition_type = "definition"
+
+        # Check whether a term is a part of entity or not.
+        for a_range in entity_ranges:
+            # check whether definiendum contains an entiity.
+            if term_start <= a_range.start and a_range.end <= term_end:
+                if term_type == "term":
+                    term_type = "entity"
+
+        # Check whether a term is a part of abbreviation or not.
+        for a_range in abbreviation_ranges:
+            # check whether definiendum contains an abbreviation.
+            if term_start <= a_range.start and a_range.end <= term_end:
+                # abbreviation over-write entitiy type because entity type is optional
+                if term_type == "term":
+                    term_type = "abbreviation"
 
         pair = TermDefinitionPair(
             term_start=term_start,
             term_end=term_end,
             term_text=text[term_start:term_end],
+            term_type=term_type,
+            term_confidence=term_confidence,
             definition_start=definition_start,
             definition_end=definition_end,
             definition_text=text[definition_start:definition_end],
+            definition_type=definition_type,
+            definition_confidence=definition_confidence
+        )
+        logging.debug("Found definition-term pair %s", pair)
+        pairs.append(pair)
+
+    return pairs
+
+
+
+def sanity_check_for_acronym_detection(
+        acronym_text, expansion_text,
+        acronym_start, acronym_end,
+        expansion_start, expansion_end):
+    """
+    Very smooth filter
+    """
```

Comment:
@dykang

Reply:
Since outputs from our current abbreviation detector include lots of noise (e.g., alpha x (Ref ), any phrases with parenthesis), I made simple heuristic rules to filter them out. I hope we can do a better job with the BERT-based model Vivek developed so we don't use this anymore in the future.

Reply:
Okay cool! Can you include that elaboration in the comment?